tests.js

```diff
@@ -16,45 +16,167 @@
 
 class Tests {
   constructor(options) {
-    this.items = options.manifest.items
-        .filter((item) => !options.httpOnly || item.protocol === 'http');
-    this.individualItems = options.manifest.individualItems;
+    this.tests = options.tests;
+    this.endpoints = this.buildEndpoints();
     this.host = options.host;
+    this.httpOnly = options.httpOnly;
   }
 
-  list(after, limit) {
-    let begin; let end;
-    if (after) {
-      const afterURL = new URL(after);
-      const afterIndex = this.items.findIndex((item) => {
-        const itemURL = new URL(`${item.protocol}://${this.host}${item.pathname}`);
-        return itemURL.pathname === afterURL.pathname &&
-            itemURL.protocol === afterURL.protocol;
-      });
+  buildEndpoints() {
+    const endpoints = {};
 
-      if (afterIndex === -1) {
-        throw new Error(`${after} not found in test manifest`);
+    for (const [ident, test] of Object.entries(this.tests)) {
+      for (const scope of test.scope) {
+        let endpoint = '';
+        let httpsOnly = false;
+        switch (scope) {
+          case 'Window':
+            endpoint = '/api/interfaces';
+            break;
+          case 'Worker':
```

Comment:
Can both 'Worker' and 'DedicatedWorker' happen? It seems like it should be one or the other.

Reply:
Some interfaces mark exposure for `Worker`, while others for `DedicatedWorker`.  From what I've gathered, they're both the same however.